vendor/riscv/riscv-isa-sim/riscv/csrs.cc

```diff
@@ -992,13 +993,15 @@ bool wide_counter_csr_t::unlogged_write(const reg_t val) noexcept {
   // takes precedence over the increment to instret.  However, Spike
   // unconditionally increments instret after executing an instruction.
   // Correct for this artifact by decrementing instret here.
-   openhw::reg::unlogged_write( openhw::reg::unlogged_read()-1);
+
+  // disable since we (at the moment) check mcounterinhibit
+  //openhw::reg::unlogged_write( openhw::reg::unlogged_read()-1);
```

Comment:
I modified this and the +1 below to add support for disabling `mcycle` counting with `mcountinhibit` to match cv32e40s. I have not tested this much with `mcountinhibit` enabled since `mcycle` would likely mismatch with the core anyway.

Reply:
Is the old behavior preserved unless the new behavior is explicitly asked for?

Reply:
I think so, but I have not done much testing with `mcountinhibit` enabled.
Since Spike "sees" fewer cycles than the core, `mcycle` would likely have a mismatch to a core anyway unless larger modifications are made.